### PR TITLE
Preserve array name in MultiIndex.from_arrays

### PR DESCRIPTION
**Pull request by mroeschke**
## Description
xref https://github.com/rapidsai/cudf/issues/16507

## Checklist
- [ ] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [ ] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.

### CODE REVIEW

File: python/cudf/cudf/tests/test_multiindex.py [hunk 1]
```diff
@@ -2179,3 +2179,10 @@ def test_unique_level():
     result = pd_mi.unique(level=1)
     expected = cudf_mi.unique(level=1)
     assert_eq(result, expected)
+
+
+def test_from_arrays_infer_names():
+    arrays = [pd.Index([1], name="foo"), pd.Index([2], name="bar")]
```
Comment (Matt711) [inline python/cudf/cudf/tests/test_multiindex.py:2185]: Do you mind adding some more test cases for different index types? I believe I saw this bug for `pd.CategoricalIndex` too.

Comment (mroeschke) [reply python/cudf/cudf/tests/test_multiindex.py]: Sure thing. Added more index classes here

**Review by Matt711 — APPROVED** [review]
Thanks! LGTM

**Pull request merged**